src/subtyping.md

```diff
@@ -59,200 +139,284 @@ To handle it, we need to talk about *variance*.
+`&mut Animal`. The static type checker will then correctly stop us from passing
+a Cat into `evil_feeder`.
 
-Some important variances (which we will explain in detail below):
+The soundness of subtyping is based on the idea that it's ok to forget unnecessary
+details. But with references, there's always someone that remembers those details:
+the value being referenced. That value expects those details to keep being true,
+and may behave incorrectly if its expectations are violated.
 
-* `&'a T` is covariant over `'a` and `T` (as is `*const T` by metaphor)
-* `&'a mut T` is covariant over `'a` but invariant over `T`
-* `fn(T) -> U` is **contra**variant over `T`, but covariant over `U`
-* `Box`, `Vec`, and all other collections are covariant over the types of
-  their contents
-* `UnsafeCell<T>`, `Cell<T>`, `RefCell<T>`, `Mutex<T>` and all other
-  interior mutability types are invariant over T (as is `*mut T` by metaphor)
+The problem with making `&mut T` covariant over `T` is that it gives us the power
+to modify the original value *when we don't remember all of its constraints*.
+And so, we can make someone have a Dog when they're certain they still have a Cat.
 
-To understand why these variances are correct and desirable, we will consider
-several examples.
+With that established, we can easily see why `&T` being covariant over `T` *is*
+sound: it doesn't let you modify the value, only look at it. Without anyway to
+mutate, there's no way for us to mess with any details. We can also see why
+UnsafeCell and all the other interior mutability types must be invariant: they
+make `&T` work like `&mut T`!
 
-We have already covered why `&'a T` should be covariant over `'a` when
-introducing subtyping: it's desirable to be able to pass longer-lived things
-where shorter-lived things are needed.
+Now what about the lifetime on references? Why is it ok for both kinds of references
+to be covariant over their lifetimes. Well, here's a two-pronged argument:
 
-Similar reasoning applies to why it should be covariant over T: it's reasonable
-to be able to pass `&&'static str` where an `&&'a str` is expected. The
-additional level of indirection doesn't change the desire to be able to pass
-longer lived things where shorter lived things are expected.
+First and foremost, subtyping references based on their lifetimes is *the entire point
+of subtyping in Rust*. The only reason we have subtyping is so we can pass
+long-lived things where short-lived things are expected. So it better work!
 
-However this logic doesn't apply to `&mut`. To see why `&mut` should
-be invariant over T, consider the following code:
+Second, and more seriously, references *own* their lifetimes, while they are
+only borrowing their referents. If you shrink down a reference's lifetime when
+you hand it to someone, that location now has a reference which owns the smaller
+lifetime. There's no way to mess with original reference's lifetime using the
+other one.
```

Comment:
I find this use of "own" somewhat hard to follow.

The way I think about it: The lifetime in a reference is just a *lower bound*. It is how long we are *sure* the thing is there, but it may be there longer and we wouldn't care. That's why making it shorter is always okay. Do you think that would work as an explanation?

Reply:
I tried rewording it a bit, wdyt?